composer/utils/checkpoint.py

```diff
@@ -694,7 +694,8 @@ def _remove_paths(obj: Union[list, dict[str, Any]], exclude_paths: list[list[str
 
     # Recurse first, so in the case of a list, the indexing is consistent
     for key, paths_to_recurse in keys_to_recurse.items():
-        _remove_paths(obj[key], paths_to_recurse)
+        if key in obj:
+            _remove_paths(obj[key], paths_to_recurse)
```

Comment:
Filed a ticket but I think this should a bug outside of the scope of this PR - the way this was written the following callback state would raise a key error because backslashes aren't respected:
```
'CheckpointSaver': {
   'all_saved_checkpoints_to_timestamp': {
      'first/ep1.pt': {'epoch': 1, 'batch': 5, 'sample': 10, 'token': 0, 'batch_in_epoch': 0, 'sample_in_epoch': 0, 'token_in_epoch': 0, 'total_wct': datetime.timedelta(microseconds=31455), 'epoch_wct': datetime.timedelta(0), 'batch_wct': datetime.timedelta(0)}
   }
}
```

Specifically our key of interest in flatterned to be
```
state/callbacks/CheckpointSaver/all_saved_checkpoints_to_timestamp/first/ep1.pt
```

So _remove_paths is look for a checkpoint state something like:
```
'CheckpointSaver': {
   'all_saved_checkpoints_to_timestamp': {
      'first': {
         'ep1.pt': {'epoch': 1, 'batch': 5, 'sample': 10, 'token': 0, 'batch_in_epoch': 0, 'sample_in_epoch': 0, 'token_in_epoch': 0, 'total_wct': datetime.timedelta(microseconds=31455), 'epoch_wct': datetime.timedelta(0), 'batch_wct': datetime.timedelta(0)}
      }
   }
}
```

Reply:
ohh yeah, i don't think you can have `/` in state dict keys (not sure if a composer thing or a general thing). structure delineation is usually done with `.`.

Reply:
Can think about this a bit more, but I don't think its great to silently skip keys that don't exist here. Can we revert this change and instead save the state in a different way? could do a `split('/')` before saving, or save a list instead of a dictionary, or something like that.